signer/src/bitcoin/client.rs

```diff
@@ -47,22 +49,22 @@ impl BitcoinInteract for ApiFallbackClient<BitcoinCoreClient> {
         &self,
         block_hash: &bitcoin::BlockHash,
     ) -> Result<Option<bitcoin::Block>, Error> {
-        self.exec(|client| async {
-            match client.inner_client().get_block(block_hash) {
-                Ok(block) => Ok(Some(block)),
-                Err(bitcoincore_rpc::Error::JsonRpc(bitcoincore_rpc::jsonrpc::Error::Rpc(
-                    RpcError { code: -5, .. },
-                ))) => Ok(None),
-                Err(error) => Err(Error::BitcoinCoreRpc(error)),
-            }
-        })
-        .await
+        self.exec(|client| async { client.get_block(block_hash) })
+            .await
     }
 
-    fn get_tx(&self, txid: &bitcoin::Txid) -> Result<GetTxResponse, Error> {
+    fn get_tx(&self, txid: &Txid) -> Result<Option<GetTxResponse>, Error> {
         self.get_client().get_tx(txid)
     }
 
+    fn get_tx_info(
+        &self,
+        txid: &Txid,
+        block_hash: &BlockHash,
+    ) -> Result<Option<BitcoinTxInfo>, Error> {
+        self.get_client().get_tx_info(txid, block_hash)
```

Comment:
Is it intentional to skip failover/retry here?

Reply:
Oh right! Yeah, it was the whole `async` thing; I tried but couldn't because `exec` is `async`. Something felt not right about forcing the trait function to be async when it doesn't need to be, and I just figured that when I revisit it the right path will be obvious.

Reply:
You know, I'll just make it all `async` and call it a day

Reply:
Oh I must have went down this path and realized that it was more work than it is worth. So yyyyeah, here we are.

Reply:
Ah, gotcha. Personally I think that all of the client trait methods should be async because we don't "know" what that underlying implementation is doing and in this case the mix/match of async feels off, like we're explicitly tying the trait to the current underlying impl.

Actually, it wouldn't be stupid to make the internals `async` with a `tokio::spawn` around the API calls since they are I/O.

Reply:
> Ah, gotcha. Personally I think that all of the client trait methods should be async because we "don't know" what that underlying implementation is doing and in this case the mix/match of async feels off, like we're explicitly tying the trait to the current underlying impl.
> 
> Actually, it wouldn't be stupid to make the internals `async` with a `tokio::spawn` around the API calls since they are indeed I/O.

Yeah, this is the better argument and I agree with it.

I did try to make the change but it was annoying and would be more time-consuming than I felt it was worth at this point.

Reply:
We can leave it for now then and attack it in some upcoming PR :)

Reply:
> Oh I must have went down this path and realized that it was more work than it is worth. So yyyyeah, here we are.

Oh for some reason I didn't see this prior to my last two replies 😅 Sorry